api/v1beta1/foundationdbcluster_types.go

```diff
@@ -1361,11 +1364,17 @@ func (cluster *FoundationDBCluster) CheckReconciliation() (bool, error) {
 	cluster.Status.Generations = ClusterGenerationStatus{Reconciled: cluster.Status.Generations.Reconciled}
 
 	for _, processGroup := range cluster.Status.ProcessGroups {
-		if processGroup.Remove && !processGroup.Excluded {
-			cluster.Status.Generations.NeedsShrink = cluster.ObjectMeta.Generation
-			reconciled = false
-		} else if processGroup.Remove {
-			cluster.Status.Generations.HasPendingRemoval = cluster.ObjectMeta.Generation
+		if processGroup.Remove {
```

Comment:
NIT: could we invert that check and just continue if the process group is not marked for deletion?